### ISSUE TITLE
Allow passing a proc/class to asset_path

### ISSUE THREAD

**Issue by stevenosloan** [middleman/middleman-sprockets#107]
This would allow users to "rebuild" the 3.x behavior of placing assets in their respective `css_dir`, `js_dir` .... In theory this should be easier for each case (since the source => dest is known) and less surprising than results could be in 3.x.

**Comment by tdreyno** [comment]
Hmm, can this use or build upon the `move_file` feature in core?

https://github.com/middleman/middleman/blob/master/middleman-core/lib/middleman-core/sitemap/extensions/move_file.rb

**Comment by stevenosloan** [comment]
Yeah, I guess it could -- I was thinking though since we're setting [linked asset paths here](https://github.com/middleman/middleman-sprockets/blob/master/lib/middleman-sprockets/extension.rb#L117-L119) that allowing users to give a proc for that would be easier to keep track of than finding those assets again and using move_file. That make sense to you?

**Comment by tdreyno** [comment]
Makes sense.

**Comment by vvasabi** [comment]
I needed this capability today, so I have created a pull request. Let me know how it looks.
